File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """

        min_price = float('inf')
        max_profit = 0
        
        for price in prices:
            #check the min_price
            if (min_price > price):
                min_price = price
            #calculate the cost
            cost = price-min_price
            #find the max_profit
            if (cost > max_profit):
                max_profit = cost
        
        return max_profit

        

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0;
        int min_buy = INT_MAX;

        for (int num: prices){
            min_buy = min(min_buy, num);
            int profit = num - min_buy;
            max_profit = max(max_profit, profit);
        }

        return max_profit;
    }
};